apps/condo/domains/meter/schema/MeterReportingPeriod.test.js

```diff
@@ -0,0 +1,269 @@
+                const unitName = faker.random.alphaNumeric(8)
+                const { context, organization } = await makeContextWithOrganizationAndIntegrationAsAdmin()
+                const [property] = await createTestProperty(admin, organization)
+                const [property2] = await createTestProperty(admin, organization)
+                const [billingProperty] = await createTestBillingProperty(admin, context)
+                const [billingAccount] = await createTestBillingAccount(admin, context, billingProperty)
+                const [resident] = await createTestResident(admin, client.user, property, {
+                    unitName,
+                })
+                await createTestServiceConsumer(admin, resident, organization, {
+                    accountNumber: billingAccount.number,
+                })
+                const [resident2] = await createTestResident(admin, client2.user, property2, {
+                    unitName,
+                })
+                await createTestServiceConsumer(admin, resident2, organization, {
+                    accountNumber: billingAccount.number,
+                })
+                await createTestMeterReportingPeriod(admin, organization, { property: { connect: { id: property.id } } })
+
+                const objs = await MeterReportingPeriod.getAll(client2, {
+                    property: { id: property.id },
+                }, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs).toHaveLength(0)
+            })
+
+            test('resident can read default MeterReportingPeriod', async () => {
+                const client = await makeClientWithResidentUser()
+                const unitName = faker.random.alphaNumeric(8)
+                const { organization } = await makeContextWithOrganizationAndIntegrationAsAdmin()
+                const [property] = await createTestProperty(admin, organization)
+                const [resident] = await createTestResident(admin, client.user, property, {
+                    unitName,
+                })
+                const defaultMeterReportingPeriod = await MeterReportingPeriod.create(admin, {
+                    dv: 1,
+                    sender: { dv: 1, fingerprint: 'initDefaultPeriod' },
+                })
+
+                const [period] = await MeterReportingPeriod.getAll(client, {
+                    organization_is_null: true,
+                }, { sortBy: ['updatedAt_DESC'] })
+
+                await MeterReportingPeriod.softDelete(admin, defaultMeterReportingPeriod.id)
+
+                expect(period.id).toEqual(defaultMeterReportingPeriod.id)
+            })
+
+            test('anonymous can\'t', async () => {
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await MeterReportingPeriod.getAll(anonymous, {}, { sortBy: ['updatedAt_DESC'] })
+                })
+            })
+        })
+    })
+
+    describe('Validation tests', () => {
```

Comment:
you should add tests for `INVALID_FINISH`, `INVALID_START`, `ORGANIZATION_IS_REQUIRED` errors